packages/react-storage/src/components/StorageBrowser/actions/handlers/download.ts

```diff
@@ -1,16 +1,42 @@
-import { TaskHandler, TaskHandlerInput, TaskHandlerOutput } from '../types';
+import { getUrl } from '../../storage-internal';
+import {
+  TaskHandler,
+  TaskHandlerInput,
+  TaskHandlerOptions,
+  TaskHandlerOutput,
+} from '../types';
+import { constructBucket, resolveHandlerResult } from './utils';
 
-interface DownloadHandlerData {
-  key: string;
-}
-interface DownloadHandlerOptions {
-  onProgress?: (progress: number) => void;
-}
+interface DownloadHandlerOptions extends TaskHandlerOptions {}
 export interface DownloadHandlerInput
-  extends TaskHandlerInput<DownloadHandlerData, DownloadHandlerOptions> {}
+  extends Omit<TaskHandlerInput<never, DownloadHandlerOptions>, 'data'> {
+  data: { key: string };
+}
 export interface DownloadHandlerOutput extends TaskHandlerOutput {}
 
 export interface DownloadHandler extends TaskHandler<DownloadHandlerInput> {}
 
-export const downloadHandler: DownloadHandler =
-  null as unknown as DownloadHandler;
+export const downloadHandler: DownloadHandler = ({
+  config,
+  data: { key },
+  prefix,
+  options,
+}): DownloadHandlerOutput => {
+  const { credentials } = config;
+  const bucket = constructBucket(config);
+
+  const result = getUrl({
+    path: `${prefix}${key}`,
+    options: {
+      bucket,
+      locationCredentialsProvider: credentials,
+      validateObjectExistence: true,
+      contentDisposition: 'attachment',
+    },
+  });
+
```

Comment:
Need to move the body of this function/run [this function](https://github.com/aws-amplify/amplify-ui/blob/feat-storage-browser/main/packages/react-storage/src/components/StorageBrowser/views/Controls/Download.tsx#L13) on success before returning so the entirety of the download logic is scoped to the handler